core/src/main/java/org/apache/calcite/util/format/PostgresqlDateTimeFormatter.java

```diff
@@ -0,0 +1,660 @@
+            if (formattedYear.length() > 3) {
+              return formattedYear.substring(formattedYear.length() - 3);
+            } else {
+              return formattedYear;
+            }
+          },
+          "YYY"),
+      new NumberFormatPattern(
+          dt -> {
+            final String formattedYear = String.format(Locale.ROOT, "%02d", dt.getYear());
+            if (formattedYear.length() > 2) {
+              return formattedYear.substring(formattedYear.length() - 2);
+            } else {
+              return formattedYear;
+            }
+          },
+          "YY"),
+      new NumberFormatPattern(
+          dt -> {
+            final String formattedYear = Integer.toString(dt.getYear());
+            if (formattedYear.length() > 1) {
+              return formattedYear.substring(formattedYear.length() - 1);
+            } else {
+              return formattedYear;
+            }
+          },
+          "Y"),
+      new NumberFormatPattern(
+          dt -> String.format(Locale.ROOT, "%02d", dt.get(IsoFields.WEEK_OF_WEEK_BASED_YEAR)),
+          "IW"),
+      new NumberFormatPattern(
+          dt -> {
+            final Month month = dt.getMonth();
+            final int dayOfMonth = dt.getDayOfMonth();
+            final int weekNumber = dt.get(IsoFields.WEEK_OF_WEEK_BASED_YEAR);
+
+            if (month == Month.JANUARY && dayOfMonth < 4) {
+              if (weekNumber == 1) {
+                return String.format(Locale.ROOT, "%03d", dt.getDayOfWeek().getValue());
+              }
+            } else if (month == Month.DECEMBER && dayOfMonth >= 29) {
+              if (weekNumber == 1) {
+                return String.format(Locale.ROOT, "%03d", dt.getDayOfWeek().getValue());
+              }
+            }
+
+            return String.format(Locale.ROOT, "%03d",
+                (weekNumber - 1) * 7 + dt.getDayOfWeek().getValue());
+          },
+          "IDDD"),
+      new NumberFormatPattern(
+          dt -> Integer.toString(dt.getDayOfWeek().getValue()),
+          "ID"),
+      new NumberFormatPattern(
+          dt -> {
+            final String yearString = Integer.toString(dt.get(IsoFields.WEEK_BASED_YEAR));
+            return yearString.substring(yearString.length() - 1);
+          },
+          "I"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "BC" : "AD",
+          "BC", "AD"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "bc" : "ad",
+          "bc", "ad"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "B.C." : "A.D.",
+          "B.C.", "A.D."),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "b.c." : "a.d.",
+          "b.c.", "a.d."),
+      new StringFormatPattern(
+          dt -> {
+            final String monthName =
+                dt.getMonth().getDisplayName(TextStyle.FULL,
+                Locale.getDefault());
```

Comment:
Are you sure about the Locale?
I thought that the spec requires the month name to be in English.

Reply:
I see what you mean. It should default to English and I should only translate when the `TM` prefix is provided. I'll make the changes.

https://www.postgresql.org/docs/14/functions-formatting.html#FUNCTIONS-FORMATTING-DATETIME-TABLE

Reply:
Updated to make use of the TM flag.